### PR TITLE
NX-507: Smoke IT added

### PR DESCRIPTION
**Pull request by cstamas**
Changes:
- added "modern" IT module with one "smoke" testing IT
- added client bits to make possible to create NPM proxy repository via client
- added mock NPM registry. The current mock registry (in smoke IT) works just fine with NPM CLI executing `npm install commonjs`

In short, to have your IT create and start a mock NPM registry, just create a folder called `registry` in `src/test/it-resources` folder in the _same package where your IT is_.

Then lay down the desired files, with metadata files (registry root, package and version) as being named `data.json` in the given directory root (so, in `registry` directory, `registry/package` or `registry/package/version` directories). The `data.json` will be filtered and served to client. Filtering is currently quite basic, only one expression is filtered for: `${npmRegistryUrl}`

### CODE REVIEW

File: nexus-npm-repository-testsuite/src/test/filtered-resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Sonatype Nexus (TM) Open Source Version
```
Comment (georgy) [inline nexus-npm-repository-testsuite/src/test/filtered-resources/logback-test.xml]: Do we want this in the plugin? :)

Comment (cstamas) [inline nexus-npm-repository-testsuite/src/test/filtered-resources/logback-test.xml]: Doh, sorry (and same for version in IT classes), wrong "template". Will remove it.

**Comment by cstamas** [comment]
To run and execute ITs, one should use following on CLI: `-Dit`. Without this, the ITs are NOT executed.

**Comment by cstamas** [comment]
About the `NpmInstallIT` test, unsure do we want to keep it, or just "record" the traffic between npm CLI and NX and "replay" that. As this IT requires npm command to be on path (windows?), and also might break if some future npm command version introduces some change not compatible with current version.... (the repo is "frozen", is mocked, so repository changes does not play here)

File: nexus-npm-repository-testsuite/src/test/filtered-resources/logback-test.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  <logger name="org.sonatype.nexus.client" level="${test.log.level.org.sonatype.nexus.client:-INFO}"/>
+
+  <logger name="org.sonatype.nexus.plugins.capabilities" level="TRACE"/>
+  <logger name="org.sonatype.nexus.capabilities" level="TRACE"/>
+  <logger name="org.sonatype.nexus.obr" level="TRACE"/>
+  <logger name="org.sonatype.nexus.unpack" level="TRACE"/>
+  <logger name="org.sonatype.nexus.repository.site" level="TRACE"/>
+  <logger name="org.sonatype.security.realms.kenai" level="TRACE"/>
+  <logger name="org.sonatype.nexus.proxy.maven.routing" level="TRACE"/>
+  <logger name="org.sonatype.nexus.yum" level="TRACE"/>
```
Comment (jdillon) [inline nexus-npm-repository-testsuite/src/test/filtered-resources/logback-test.xml:23]: why is all this stuff at TRACE?

**Pull request merged**